### PR TITLE
Create geography_type enum and add geography column in connection table

### PR DESCRIPTION
**Pull request by terencecho**
## What
In preparation to support a routing service for jobs to be put in different task queues based off the data planes users want to run them in, this PR creates a geography enum type in the database (currently populated with `AUTO`, `US` and `EU`) and creates a `geography` column in the connection table.

### CODE REVIEW

**Comment by terencecho** [comment]
todo: edit name of file to reflect the current oss version

**Review by pmossman — APPROVED** [review]
Sorry I took a while to get to this, looks good!

File: airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_40_6_001__AddGeographyColumnToConnections.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    addGeographyColumn(ctx);
+  }
+
+  private static void addGeographyEnumDataTypes(final DSLContext ctx) {
+    ctx.createType(GeographyType.NAME)
+        .asEnum(Arrays.stream(GeographyType.values()).map(GeographyType::getLiteral).toList())
+        .execute();
+  }
+
+  private static void addGeographyColumn(final DSLContext ctx) {
```
Comment (pmossman) [inline airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_40_6_001__AddGeographyColumnToConnections.java:43]: @terencecho just realized, I think we want all existing connection record to be set to 'AUTO' during this migration, right?

**Comment by pmossman** [comment]
@terencecho I started on the API changes that will use this, and I realized we might want to make some changes to this migration before we merge/run it in prod.

- I think it probably makes sense to just add the column to the workspace table as well as connections in a single migration,  I think it isn't much of a lift and we might as well just knock out both changes in one migration since they're conceptually related
- I think we should make the column explicitly non-nullable, and declare a default value of 'auto' in the migration so that existing workspaces and connection records get set to 'auto' during the migration

Let me know if any of that doesn't make sense or if you wanna chat about it!

**Comment by terencecho** [comment]
@pmossman sounds good to me. i can make those changes on tuesday.

**Comment by pmossman** [comment]
New updates since my last review look good, consider this re-approved!

**Pull request merged**